src/sql/workbench/browser/modal/modal.ts

```diff
@@ -316,29 +317,30 @@ export abstract class Modal extends Disposable implements IThemable {
 	}
 
 	/**
-	 * Set focusable elements in the modal dialog
+	 * Figures out the first and last elements which the user can tab to in the dialog
 	 */
-	public setFocusableElements() {
-		// try to find focusable element in dialog pane rather than overall container
-		this._focusableElements = this._modalBodySection ?
-			this._modalBodySection.querySelectorAll('a[href], area[href], input:not([disabled]), select:not([disabled]), textarea:not([disabled]), button:not([disabled]), [tabindex="0"]') :
-			this._bodyContainer.querySelectorAll('a[href], area[href], input:not([disabled]), select:not([disabled]), textarea:not([disabled]), button:not([disabled]), [tabindex="0"]');
-		if (this._focusableElements && this._focusableElements.length > 0) {
-			this._firstFocusableElement = <HTMLElement>this._focusableElements[0];
-			this._lastFocusableElement = <HTMLElement>this._focusableElements[this._focusableElements.length - 1];
+	public setFirstLastTabbableElement() {
+		let tabbableElements = this._bodyContainer.querySelectorAll('a[href], area[href], input:not([disabled]), select:not([disabled]), textarea:not([disabled]), button:not([disabled]), [tabindex="0"]');
+		if (tabbableElements && tabbableElements.length > 0) {
+			this._firstTabbableElement = <HTMLElement>tabbableElements[0];
+			this._lastTabbableElement = <HTMLElement>tabbableElements[tabbableElements.length - 1];
 		}
-
-		this._focusedElementBeforeOpen = <HTMLElement>document.activeElement;
-		this.focus();
 	}
 
 	/**
-	 * Focuses the modal
-	 * Default behavior: focus the first focusable element
+	 * Set focusable elements in the modal dialog
 	 */
-	protected focus() {
-		if (this._firstFocusableElement) {
-			this._firstFocusableElement.focus();
+	public setFocusableElements() {
+		// Try to find focusable element in dialog pane rather than overall container. _modalBodySection contains items in the pane for a wizard.
+		// This ensures that we are setting the focus on a useful element in the form when possible.
+		let focusableElements = this._modalBodySection ?
+		this._modalBodySection.querySelectorAll('a[href], area[href], input:not([disabled]), select:not([disabled]), textarea:not([disabled]), button:not([disabled]), [tabindex="0"]') :
+		this._bodyContainer.querySelectorAll('a[href], area[href], input:not([disabled]), select:not([disabled]), textarea:not([disabled]), button:not([disabled]), [tabindex="0"]');
+
+		this._focusedElementBeforeOpen = <HTMLElement>document.activeElement;
+
+		if (focusableElements[0]) {
```

Comment:
Just to be safe should check focusableElements.length > 0. (I wouldn't' expect that to ever happen though since we have the buttons - and I'm not sure what we would do in that case if we did indeed have no focusable elements. Probably just capture the tab events and stop them propagating but not do anything else?)

Reply:
Good point. I'll add that check.